File: infra-market-server-go/internal/dto/activity.go
package dto

// ActivityDto 活动信息DTO
type ActivityDto struct {
	ID           uint64                `json:"id"`
	Name         string                `json:"name"`
	Description  *string               `json:"description"`
	TemplateID   uint64                `json:"templateId"`
	TemplateName *string               `json:"templateName"`
	ConfigData   map[string]any `json:"configData"`
	Status       int                   `json:"status"`
	CreateTime   string                `json:"createTime"`
	UpdateTime   string                `json:"updateTime"`
}

// ActivityFormDto 活动创建/更新表单
type ActivityFormDto struct {
	Name        string                 `json:"name" binding:"required,min=1,max=100"`
	Description *string                `json:"description" binding:"omitempty,max=500"`
	TemplateID  uint64                 `json:"templateId" binding:"required,min=1"`
	ConfigData  map[string]any `json:"configData"`
	Status      *int                   `json:"status" binding:"omitempty,oneof=0 1"`
}

// ActivityQueryDto 活动查询DTO
type ActivityQueryDto struct {
	Name       *string `form:"name"`
	TemplateID *uint64 `form:"templateId"`
	Status     *int    `form:"status" binding:"omitempty,oneof=0 1"`
	Pagination
}

// ActivityTemplateDto 活动模板信息DTO
type ActivityTemplateDto struct {
	ID          uint64                      `json:"id"`
	Name        string                      `json:"name"`
	Description *string                     `json:"description"`
	Fields      []ActivityTemplateFieldDto   `json:"fields"`
	Status      int                         `json:"status"`
	CreateTime  string                      `json:"createTime"`
	UpdateTime  string                      `json:"updateTime"`
}

// ActivityTemplateFieldDto 活动模板字段配置DTO
type ActivityTemplateFieldDto struct {
	Name         string                           `json:"name"`
	Label        *string                          `json:"label"`
	Type         *string                          `json:"type"`
	InputType    *string                          `json:"inputType"`
	Required     *bool                            `json:"required"`
	DefaultValue any                      `json:"defaultValue"`
	Description  *string                          `json:"description"`
	Sort         *int                             `json:"sort"`
	Options      []SelectOptionDto                `json:"options"`
	Multiple     *bool                            `json:"multiple"`
	Min          any                      `json:"min"`
	Max          any                      `json:"max"`
	MinLength    *int                             `json:"minLength"`
	MaxLength    *int                             `json:"maxLength"`
	Pattern      *string                          `json:"pattern"`
	Placeholder  *string                          `json:"placeholder"`
	ItemType     *ActivityTemplateFieldDto         `json:"itemType"`
	Properties   map[string]ActivityTemplateFieldDto `json:"properties"`
	ComponentID  *uint64                          `json:"componentId"`
	IsArray      *bool                            `json:"isArray"`
	AllowDynamic *bool                            `json:"allowDynamic"`
}

// ActivityTemplateFormDto 活动模板创建/更新表单
type ActivityTemplateFormDto struct {
	Name        string                      `json:"name" binding:"required,min=1,max=100"`
	Description *string                     `json:"description" binding:"omitempty,max=500"`
	Fields      []ActivityTemplateFieldDto   `json:"fields"`
	Status      *int                        `json:"status" binding:"omitempty,oneof=0 1"`
}

// ActivityTemplateQueryDto 活动模板查询DTO
type ActivityTemplateQueryDto struct {
	Name   *string `form:"name"`
	Status *int    `form:"status" binding:"omitempty,oneof=0 1"`
	Pagination
}

// ActivityComponentDto 活动组件信息DTO
type ActivityComponentDto struct {
	ID          uint64                      `json:"id"`
	Name        string                      `json:"name"`
	Description *string                     `json:"description"`
	Fields      []ActivityComponentFieldDto  `json:"fields"`
	Status      int                         `json:"status"`
	CreateTime  string                      `json:"createTime"`
	UpdateTime  string                      `json:"updateTime"`
}

// ActivityComponentFieldDto 活动组件字段/组件配置DTO
type ActivityComponentFieldDto struct {
	Name         string                           `json:"name"`
	Label        *string                          `json:"label"`
	Type         *string                          `json:"type"`
	InputType    *string                          `json:"inputType"`
	Required     *bool                            `json:"required"`
	DefaultValue any                      `json:"defaultValue"`
	Description  *string                          `json:"description"`
	Sort         *int                             `json:"sort"`
	Options      []SelectOptionDto                `json:"options"`
	Multiple     *bool                            `json:"multiple"`
	Min          any                      `json:"min"`
	Max          any                      `json:"max"`
	MinLength    *int                             `json:"minLength"`
	MaxLength    *int                             `json:"maxLength"`
	Pattern      *string                          `json:"pattern"`
	Placeholder  *string                          `json:"placeholder"`
	ItemType     *ActivityComponentFieldDto       `json:"itemType"`
	Properties   map[string]ActivityComponentFieldDto `json:"properties"`
	ComponentID  *uint64                          `json:"componentId"`
	IsArray      *bool                            `json:"isArray"`
	AllowDynamic *bool                            `json:"allowDynamic"`
}

// ActivityComponentFormDto 活动组件创建/更新表单
type ActivityComponentFormDto struct {
	Name        string                      `json:"name" binding:"required,min=1,max=100"`
	Description *string                     `json:"description" binding:"omitempty,max=500"`
	Fields      []ActivityComponentFieldDto  `json:"fields"`
	Status      *int                        `json:"status" binding:"omitempty,oneof=0 1"`
}

// ActivityComponentQueryDto 活动组件查询DTO
type ActivityComponentQueryDto struct {
	Name   *string `form:"name"`
	Status *int    `form:"status" binding:"omitempty,oneof=0 1"`
	Pagination
}

File: infra-market-server-dotnet/DTOs/Activity.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json;

namespace InfraMarket.Server.DTOs;

public class ActivityDto
{
    public ulong Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public ulong TemplateId { get; set; }
    public string? TemplateName { get; set; }
    public Dictionary<string, object>? ConfigData { get; set; }
    public int Status { get; set; }
    public string CreateTime { get; set; } = string.Empty;
    public string UpdateTime { get; set; } = string.Empty;
}

public class ActivityFormDto
{
    [Required]
    [MinLength(1)]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Description { get; set; }

    [Required]
    [Range(1, ulong.MaxValue)]
    public ulong TemplateId { get; set; }

    public Dictionary<string, object>? ConfigData { get; set; }

    [Range(0, 1)]
    public int? Status { get; set; }
}

public class ActivityQueryDto
{
    public string? Name { get; set; }
    public ulong? TemplateId { get; set; }
    [Range(0, 1)]
    public int? Status { get; set; }
    public int? Page { get; set; }
    public int? Size { get; set; }
}

public class ActivityTemplateDto
{
    public ulong Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public List<ActivityTemplateFieldDto>? Fields { get; set; }
    public int Status { get; set; }
    public string CreateTime { get; set; } = string.Empty;
    public string UpdateTime { get; set; } = string.Empty;
}

public class ActivityTemplateFieldDto
{
    public string Name { get; set; } = string.Empty;
    public string? Label { get; set; }
    public string? Type { get; set; }
    public string? InputType { get; set; }
    public bool? Required { get; set; }
    public object? DefaultValue { get; set; }
    public string? Description { get; set; }
    public int? Sort { get; set; }
    public List<SelectOptionDto>? Options { get; set; }
    public bool? Multiple { get; set; }
    public object? Min { get; set; }
    public object? Max { get; set; }
    public int? MinLength { get; set; }
    public int? MaxLength { get; set; }
    public string? Pattern { get; set; }
    public string? Placeholder { get; set; }
    public ActivityTemplateFieldDto? ItemType { get; set; }
    public Dictionary<string, ActivityTemplateFieldDto>? Properties { get; set; }
    public ulong? ComponentId { get; set; }
    public bool? IsArray { get; set; }
    public bool? AllowDynamic { get; set; }
}

public class ActivityTemplateFormDto
{
    [Required]
    [MinLength(1)]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Description { get; set; }

    public List<ActivityTemplateFieldDto>? Fields { get; set; }

    [Range(0, 1)]
    public int? Status { get; set; }
}

public class ActivityTemplateQueryDto
{
    public string? Name { get; set; }
    [Range(0, 1)]
    public int? Status { get; set; }
    public int? Page { get; set; }
    public int? Size { get; set; }
}

public class ActivityComponentDto
{
    public ulong Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public List<ActivityComponentFieldDto>? Fields { get; set; }
    public int Status { get; set; }
    public string CreateTime { get; set; } = string.Empty;
    public string UpdateTime { get; set; } = string.Empty;
}

public class ActivityComponentFieldDto
{
    public string Name { get; set; } = string.Empty;
    public string? Label { get; set; }
    public string? Type { get; set; }
    public string? InputType { get; set; }
    public bool? Required { get; set; }
    public object? DefaultValue { get; set; }
    public string? Description { get; set; }
    public int? Sort { get; set; }
    public List<SelectOptionDto>? Options { get; set; }
    public bool? Multiple { get; set; }
    public object? Min { get; set; }
    public object? Max { get; set; }
    public int? MinLength { get; set; }
    public int? MaxLength { get; set; }
    public string? Pattern { get; set; }
    public string? Placeholder { get; set; }
    public ActivityComponentFieldDto? ItemType { get; set; }
    public Dictionary<string, ActivityComponentFieldDto>? Properties { get; set; }
    public ulong? ComponentId { get; set; }
    public bool? IsArray { get; set; }
    public bool? AllowDynamic { get; set; }
}

public class ActivityComponentFormDto
{
    [Required]
    [MinLength(1)]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Description { get; set; }

    public List<ActivityComponentFieldDto>? Fields { get; set; }

    [Range(0, 1)]
    public int? Status { get; set; }
}

public class ActivityComponentQueryDto
{
    public string? Name { get; set; }
    [Range(0, 1)]
    public int? Status { get; set; }
    public int? Page { get; set; }
    public int? Size { get; set; }
}
